### PR TITLE
Limit maximum number of cases in case server

### PR DESCRIPTION
**Pull request by JacquesBorsenberger**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*

**What is the current behavior?** *(You can also link to an open issue here)*

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: case-server/src/main/java/com/powsybl/caseserver/CaseService.java [hunk 1]
```diff
@@ -66,6 +68,9 @@
     @Lazy
     private CaseInfosService caseInfosService;
 
+    @Value("#{${max-public-cases:systemProperties['maxPublicCases']?: '-1'}}")
```
Comment (jonenst) [inline case-server/src/main/java/com/powsybl/caseserver/CaseService.java:71]: why `:systemProperties['maxPublicCases']?` ?

also it's better to  just put the default value -1 in the shipped application.yml than here in the code ?

**Comment by jonenst** [comment]
even if this is not the best system to implement a rolling storage, it's simple and will be improved when we switch to object storage, so I think this is good.

File: case-server/src/main/java/com/powsybl/caseserver/CaseService.java [hunk 2]
```diff
@@ -213,6 +219,33 @@ UUID importCase(MultipartFile mpf, boolean toPublic) {
...
+        if (capacity < 0) {
+            return;
+        }
+        List<Path> listFiles = new ArrayList<>();
+        try (DirectoryStream<Path> paths = Files.newDirectoryStream(directory)) {
+            paths.forEach(listFiles::add);
+        } catch (IOException e) {
+            LOGGER.error(e.getMessage());
+        }
+        if (listFiles.size() > capacity) {
```
Comment (FranckLecuyer) [inline case-server/src/main/java/com/powsybl/caseserver/CaseService.java:232]: shouldn't it be listFiles.size() >= capacity ?

File: case-server/src/main/java/com/powsybl/caseserver/CaseService.java [hunk 3]
```diff
@@ -66,6 +68,9 @@
     @Lazy
     private CaseInfosService caseInfosService;
 
+    @Value("#{${max-public-cases:'-1'}}")
```
Comment (jonenst) [inline case-server/src/main/java/com/powsybl/caseserver/CaseService.java:71]: in other places we use for exemple:
```
@Value("${notification.websocket.heartbeat.interval:30}") int heartbeatInterval
```
So the `#{` is useless ?

**Review by jonenst — APPROVED** [review]

**Pull request merged**